src/aktualizr_info/main.cc

```diff
@@ -77,219 +78,237 @@ int main(int argc, char **argv) {
+
+          auto load_installed_version_res = storage->loadInstalledVersions((it->first).ToString(), &installed_targets,
+                                                                           &current_version, &pending_version);
+
+          if (!load_installed_version_res ||
+              (current_version >= installed_targets.size() && pending_version >= installed_targets.size())) {
+            std::cout << "   no details about installed nor pending images\n";
+          } else {
+            if (installed_targets.size() > current_version) {
+              std::cout << "   installed image hash: " << installed_targets[current_version].sha256Hash() << "\n";
+              std::cout << "   installed image filename: " << installed_targets[current_version].filename() << "\n";
+            }
+            if (installed_targets.size() > pending_version) {
+              std::cout << "   pending image hash: " << installed_targets[pending_version].sha256Hash() << "\n";
+              std::cout << "   pending image filename: " << installed_targets[pending_version].filename() << "\n";
+            }
           }
         }
       }
-    }
 
-    std::vector<MisconfiguredEcu> misconfigured_ecus;
-    storage->loadMisconfiguredEcus(&misconfigured_ecus);
-    if (misconfigured_ecus.size() != 0u) {
-      std::cout << "Removed or not registered ecus:" << std::endl;
-      std::vector<MisconfiguredEcu>::const_iterator it;
-      for (it = misconfigured_ecus.begin(); it != misconfigured_ecus.end(); ++it) {
-        std::cout << "   '" << it->serial << "' with hardware_id '" << it->hardware_id << "' "
-                  << (it->state == EcuState::kOld ? "has been removed from config" : "not registered yet") << std::endl;
+      std::vector<MisconfiguredEcu> misconfigured_ecus;
+      storage->loadMisconfiguredEcus(&misconfigured_ecus);
+      if (misconfigured_ecus.size() != 0u) {
+        std::cout << "Removed or not registered ecus:" << std::endl;
+        std::vector<MisconfiguredEcu>::const_iterator it;
+        for (it = misconfigured_ecus.begin(); it != misconfigured_ecus.end(); ++it) {
+          std::cout << "   '" << it->serial << "' with hardware_id '" << it->hardware_id << "' "
+                    << (it->state == EcuState::kOld ? "has been removed from config" : "not registered yet")
+                    << std::endl;
+        }
       }
-    }
 
-    auto registered = storage->loadEcuRegistered();
-    std::cout << "Provisioned on server: " << (registered ? "yes" : "no") << std::endl;
-    std::cout << "Fetched metadata: " << (has_metadata ? "yes" : "no") << std::endl;
+      auto registered = storage->loadEcuRegistered();
+      std::cout << "Provisioned on server: " << (registered ? "yes" : "no") << std::endl;
+      std::cout << "Fetched metadata: " << (has_metadata ? "yes" : "no") << std::endl;
+      provision_trigger = (provision_trigger && registered);
+      provision_trigger = (provision_trigger && has_metadata);
```

Comment:
These can be merged into one line: `provisioned = (registered && has_metadata);`